scapy/contrib/bgp.py

```diff
@@ -181,13 +180,12 @@ def addfield(self, pkt, s, val):
         return s + self.i2m(pkt, val)
 
     def getfield(self, pkt, s):
-        length = self.mask2iplen(struct.unpack(">B", s[0])[0]) + 1
+        length = self.mask2iplen(orb(s[0])) + 1
         return s[length:], self.m2i(pkt, s[:length])
 
     def m2i(self, pkt, m):
-        mask = struct.unpack(">B", m[0])[0]
-        ip = "".join(
-            [m[i + 1] if i < self.mask2iplen(mask) else b"\x00" for i in range(16)])
+        mask = orb(m[0])
+        ip = b"".join(m[i + 1] if i < self.mask2iplen(mask) else b"\x00" for i in range(16))
```

Comment:
Will this work with Python 3? I think you need `m[i + 1:i + 2]`, or did I miss something?